Automattic/wp-calypso
# Update experimental features tests

NOTE: This PR is currently targeted at the `add/experimental-block-patterns-tests` branch from #48032. Once that PR is merged, we'll rebase against that new trunk instead.

This PR is a followup to #48032 that fixes the experimental features tests and combines the experimental data tests introduced in #48032 together with them in the one file.

There were a few things that were preventing the experimental feature tests from running successfully:
- The tests need the `@parallel` tag to get picked up on CircleCI (although they run fine locally without them)
- The javascript passed to the selenium `driver.executeScript()` needs a `return` statement. The driver was returning `null` no matter what happened in the test environment, leading to false positives.
- A typo that caused another assert to always pass.

After fixing those, we still see a failure with '__experimentalMainDashboardButton', and I think this might be a legitimate failure, as I can't see `interface` or `__experimentalMainDashboardButton` on the global `wp` property in the editor.

#### Testing instructions
You can re-trigger the tests to run on CircleCI by removing and re-adding the `[Status] Needs e2e Testing` label, but if you want to test changes there (like removing the failing check to test the rest) you'll need to push a new branch and add the label there.

You can also run the tests locally from the `test/e2e` directory in calypso using something like:
`env BROWSERSIZE=desktop npx mocha specs/wp-gutenberg-experimental-features-spec.js`
(You may or may not need some config decryption magic as described in the Field guide "Automated end-to-end Testing" page)

Also fixes #47839.
